### ISSUE TITLE
Why the team not work to fix the big issue for dart 3 

### ISSUE THREAD

**Issue by the-best-is-best** [onepub-dev/dcli#231]
Synchronous waiting using dart:cli waitFor and C API Dart_WaitForEvent is deprecated and disabled by default. This feature will be fully removed in Dart 3.4 release. You can currently still enable it by passing --enable_deprecated_wait_for to the Dart VM. See https://dartbug.com/52121.

**Comment by bsutton** [comment]
It is been actively worked on.

The 4.x breach which is in Alpha is where the work is been done.

The alpha is already published to pub.dev.

**Comment by the-best-is-best** [comment]
Future<void> main() async {
  var future = startProcess();

  print('****************waitforMe *****************');
  waitForMe(future);

  future = startProcess();

  await waitForEx(future);
} @bsutton  why this code in dart 3.3.0 not work

**Comment by the-best-is-best** [comment]
This package not work with dart 3.3.0 yet

**Comment by fotiDim** [comment]
Can we have a release for 4.0.1-alpha.10? 4.0.1-alpha.8 which is the current one still has traces of waitFor.

**Comment by bsutton** [comment]
Will be out in the next 24 hours

**Comment by fotiDim** [comment]
@bsutton thanks for the release. It is out but it seems you forgot dcli_terminal which is still in 4.0.1-alpha.9. I am getting this error:
```
Resolving dependencies... (1.3s)
Because dcli >=4.0.1-alpha.11 depends on dcli_terminal ^4.0.1-alpha.11 which doesn't match any versions, dcli >=4.0.1-alpha.11 is forbidden.
So, because flutter_translation_sheet depends on dcli ^4.0.1-alpha.11, version solving failed.
```

**Comment by bsutton** [comment]
The release had some issues.

I'm trying to resolve them and will look to push another release tomorrow.

**Comment by bsutton** [comment]
dcli 4.0.1-beta.1 has been released

**Comment by bsutton** [comment]
closed as duplciate of #229
